program/src/processor.rs

```diff
@@ -4734,7 +4984,9 @@ impl Processor {
         mango_account.mango_group = *mango_group_ai.key;
         mango_account.owner = *owner_ai.key;
         mango_account.order_market = [FREE_ORDER_SLOT; MAX_PERP_OPEN_ORDERS];
-        mango_account.meta_data = MetaData::new(DataType::MangoAccount, 0, true);
+        mango_account.meta_data = MetaData::new(DataType::MangoAccount, 1, true);
+
+        mango_group.num_mango_accounts += 1;
```

Comment:
The `// TODO - test passing in MangoAccount that already has some data in it` seems unsure if `seed_and_create_pda()` will fail when called on an account that has already been initialized. I think it will, because `system_instruction::allocate()` will bail on already-allocated accounts.

Nevertheless, please double check that. If one could call `create_mango_account` multiple times on the same account, one could inflate num_mango_accounts easily.

If it's safe, remove the TODO comment in all places it exists.

Reply:
Yes, I was also unsure, @riordanp could you create a test that does this?

Reply:
Tested this, calling create twice fails with this error the second time:
```
Allocate: account Address { address: ..., base: None } already in use
```